Update bin/add-exercise

keiravillekode opened the pull request:
Updated from [generic-track](https://github.com/exercism/generic-track/blob/main/bin/add-practice-exercise)

We extend the script to sort exercises by (difficulty, slug)

kotp reviewed (changes requested):
The title and topic of this seems to be "Update bin/add-exercise" but it has more than only that.

The restructuring of the order of the exercises is a completely different topic.

These should be two different pull requests, right?

If this had only the adjustment of the difficulties, I would have approved it and we would be done with that.  But because it is attached to this topic, it is going to be more convoluted.

It is a good thing that the commits are atomic, though, so this is easily remedied.

keiravillekode commented:
I added sorting to bin/add-exercise, so then I also sort config.json  But then some problematic difficulty estimates become obvious (anagram should not be 1). 

Now extracted the difficulty adjustment into https://github.com/exercism/zig/pull/453

From there, the config.json in this PR can be recreated using

```
filter='.exercises.practice = (.exercises.practice | sort_by(.difficulty, .slug))'
jq "${filter}" config.json > config.sorted && mv config.sorted config.json
```

kotp commented:
So this patch should likely only involve commit d461540 right?
